dvc/path/__init__.py

```diff
@@ -0,0 +1,85 @@
+import os
+
+from dvc.utils.compat import urlparse
+
+
+class BasePathInfo(object):
+    scheme = None
+
+    def __init__(self, path, url):
+        self.path = path
+        self.url = url
+
+    def __str__(self):
+        return self.url
+
+
+class SSHPathInfo(BasePathInfo):
+    scheme = "ssh"
+
+    def __init__(self, host, user, port, path, url):
```

Comment:
Maybe it is worth using keyword arguments everywhere? Should avoid the confusion. Also, it feels like `url` should be the first argument since it is the most generic one :slightly_smiling_face: